packages/jellyfish-api-core/__tests__/category/oracle/listLatestRawPrices.test.ts

```diff
@@ -0,0 +1,255 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import BigNumber from 'bignumber.js'
+import { OracleRawPriceState } from '../../../src/category/oracle'
+
+describe('Oracle', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should listLatestRawPrices', async () => {
+    const priceFeed1 = [
+      { token: 'APPLE', currency: 'EUR' },
+      { token: 'APPLE', currency: 'USD' }
+    ]
+    const priceFeed2 = [
+      { token: 'TESLA', currency: 'EUR' },
+      { token: 'TESLA', currency: 'USD' }
+    ]
+
+    const oracleid1 = await container.call('appointoracle', [await container.getNewAddress(), priceFeed1, 1])
+    const oracleid2 = await container.call('appointoracle', [await container.getNewAddress(), priceFeed2, 2])
+
+    await container.generate(1)
+
+    const timestamp1 = Math.floor(new Date().getTime() / 1000)
+    const timestamp2 = Math.floor(new Date().getTime() / 1000) + 300
+
+    const prices1 = [{ tokenAmount: '0.5@APPLE', currency: 'EUR' }]
+    await container.call('setoracledata', [oracleid1, timestamp1, prices1])
+
+    const prices2 = [{ tokenAmount: '1@APPLE', currency: 'USD' }]
+    await container.call('setoracledata', [oracleid1, timestamp1, prices2])
+
+    const prices3 = [{ tokenAmount: '1.5@TESLA', currency: 'EUR' }]
+    await container.call('setoracledata', [oracleid2, timestamp2, prices3])
+
+    const prices4 = [{ tokenAmount: '2@TESLA', currency: 'USD' }]
+    await container.call('setoracledata', [oracleid2, timestamp2, prices4])
+
+    await container.generate(1)
+
+    // NOTE(jingyi2811): Pagination is not supported.
+    const data = await client.oracle.listLatestRawPrices()
+
+    const result1 = data.filter(element => element.oracleid === oracleid1)
+
+    expect(result1).toStrictEqual(
+      [
+        {
+          priceFeeds: priceFeed1[0],
```

Comment:
```suggestion
          priceFeeds: { token: 'APPLE', currency: 'EUR' },
```
I prefer explicitly rather than referencing.